src/context/AuthContext.tsx

```diff
@@ -0,0 +1,47 @@
+import { User } from 'lucide-react';
+import { createContext, ReactNode, useState } from 'react';
+
+import type { UserType } from '../repository/localstorage';
+import { UserLocalStorageRepository } from '../repository/localstorage';
+
+interface User {
+  id: string;
+  name: string;
+  email: string;
+  token: string;
+}
+
+interface AuthContextType {
+  user: User | null;
+  login: (user: User) => void;
+  logout: () => void;
```

Comment:
login은 유저의 정보를 저장하는 행위, logout은 유저의 정보를 삭제하는 행위입니다. 이때 꼭 로그인, 로그아웃일 때만 유저 정보의 수정이 발생할까요? 만약 토큰이 만료되어 재발급을 받게 된다면 login 행위가 아님에서 login이라는 네이밍을 사용하게 된다는 문제가 있을 수 있습니다. 저는 보통 이러한 setter 함수들을 다른 기능에 종속되지 않게 이런 식으로 지정합니다.

```suggestion
  setUser: (user: User) => void;
  clearUser: () => void;
```